src/app/content/components/TableOfContents/index.tsx

```diff
@@ -23,28 +24,185 @@ interface SidebarProps {
   page?: Page;
 }
 
+function TabTrapper({
+  mRef,
+  isTocOpen,
+}: {
+  mRef: MutableRefObject<HTMLElement>;
+  isTocOpen: boolean;
+}) {
+  const isMobile = useMatchMobileMediumQuery();
+
+  React.useEffect(() => {
+    if (!isMobile) {
+      return;
+    }
+    const listener = createTrapTab(mRef);
+
+    if (isTocOpen) {
+      document?.addEventListener('keydown', listener, true);
+    } else {
+      document?.removeEventListener('keydown', listener, true);
+    }
+
+    return () => document?.removeEventListener('keydown', listener, true);
+  }, [mRef, isTocOpen, isMobile]);
+
+  return null;
+}
+
+
+
 // tslint:disable-next-line:variable-name
-const SidebarBody = React.forwardRef<HTMLElement, React.ComponentProps<typeof SidebarPaneBody>>(
-  (props, ref) => <SidebarPaneBody
-    ref={ref}
-    data-testid='toc'
-    aria-label={useIntl().formatMessage({id: 'i18n:toc:title'})}
-    data-analytics-region='toc'
-    {...props}
-  />
-);
+const SidebarBody = React.forwardRef<
+  HTMLElement,
+  React.ComponentProps<typeof SidebarPaneBody>
+>((props, ref) => {
+  const mRef = ref as MutableRefObject<HTMLElement>;
+
+  React.useEffect(
+    () => {
+      const firstItemInToc = mRef?.current?.querySelector(
+        'ol > li a, old > li summary'
+      ) as HTMLElement;
+      const el = mRef.current;
+      const transitionListener = () => {
+        firstItemInToc?.focus();
+      };
+
+      if (props.isTocOpen) {
+        // This is primarily for code coverage
+        transitionListener();
```

Comment:
is this still necessary since you figured out the event thing?

Reply:
Yes, `transitionListener` otherwise doesn't get called because the `transitionEnd` event on the element doesn't happen in testing.